### PR TITLE
feat: info confirmed false completion

### PR DESCRIPTION
**Pull request by sshanzel**
### Describe what this PR does
- Replace the redirect mechanism to display a modal to update missing information.

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

WT-405 #done

### CODE REVIEW

**Review by rebelchris — CHANGES_REQUESTED** [review]
I guess this will work, but only on mainLayout.
Can we not capture it in the already opened AuthModal since that won't close?

**Comment by sshanzel** [comment]
> Can we not capture it in the already opened AuthModal since that won't close?

The auth modal I believe closes when the user data is present. So I think we can't use the AuthModal.

File: packages/shared/src/components/MainLayout.tsx [hunk 1]
```diff
@@ -235,6 +247,19 @@ export default function MainLayout({
           onRequestClose={() => setIsIncompleteRegistration(false)}
         />
       )}
+      {user && !user.infoConfirmed && (
+        <StyledModal
+          isOpen
+          onRequestClose={() => setIsIncompleteRegistration(false)}
```
Comment (rebelchris) [inline packages/shared/src/components/MainLayout.tsx:253]: You can actually simply log them out here. It's most likely a ghost user, and they should not be able to continue browsing.

**Review by rebelchris — CHANGES_REQUESTED** [review]
Just one more thing.

File: packages/shared/src/components/MainLayout.tsx (hunk 1 above)
Comment (sshanzel) [reply packages/shared/src/components/MainLayout.tsx:253]: Ah, true. We should just do that. Thanks for the note.

File: packages/shared/src/components/MainLayout.tsx [hunk 2]
```diff
@@ -235,6 +247,16 @@ export default function MainLayout({
           onRequestClose={() => setIsIncompleteRegistration(false)}
         />
       )}
+      {user && !user.infoConfirmed && (
+        <StyledModal isOpen onRequestClose={() => logout()}>
+          <SocialRegistrationForm
+            provider=""
```
Comment (rebelchris) [inline packages/shared/src/components/MainLayout.tsx:253]: Maybe just best to make it optional then?

**Review by rebelchris — APPROVED** [review]
Just one little note :)

File: packages/shared/src/components/MainLayout.tsx (hunk 2 above)
Comment (sshanzel) [reply packages/shared/src/components/MainLayout.tsx:253]: Totally agree with that.

**Pull request merged**